src/acp/policy.ts

```diff
@@ -0,0 +1,180 @@
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { acp, rdf } from "../constants";
+import {
+  internal_toIriString,
+  SolidDataset,
+  Thing,
+  ThingPersisted,
+  Url,
+  UrlString,
+} from "../interfaces";
+import { internal_defaultFetchOptions } from "../resource/resource";
+import {
+  createSolidDataset,
+  saveSolidDatasetAt,
+} from "../resource/solidDataset";
+import { getUrl, getUrlAll } from "../thing/get";
+import { setUrl } from "../thing/set";
+import {
+  createThing,
+  getThing,
+  getThingAll,
+  isThingLocal,
+  removeThing,
+  setThing,
+} from "../thing/thing";
+
+export type PolicyDataset = SolidDataset;
+export type Policy = ThingPersisted;
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Initialise a new empty [[SolidDataset]] to story [[Policy]]'s in.
+ */
+export const createPolicyDataset = createSolidDataset;
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Mark a given [[SolidDataset]] as containing [[Policy]]'s, and save it to the given URL.
+ *
+ * @param url URL to save this Access Policy SolidDataset at.
+ * @param dataset The SolidDataset containing Access Policies to save.
+ * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
+ */
+export async function savePolicyDatasetAt(
+  url: Url | UrlString,
+  dataset: PolicyDataset,
+  options: Partial<
+    typeof internal_defaultFetchOptions
+  > = internal_defaultFetchOptions
+): ReturnType<typeof saveSolidDatasetAt> {
+  url = internal_toIriString(url);
+  let datasetThing = getThing(dataset, url) ?? createThing({ url: url });
+  datasetThing = setUrl(datasetThing, rdf.type, acp.AccessPolicyResource);
+  dataset = setThing(dataset, datasetThing);
+
+  return saveSolidDatasetAt(url, dataset, options);
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Initialise a new, empty [[Policy]].
+ *
+ * @param url URL that identifies this Access Policy.
+ */
+export function createPolicy(url: Url | UrlString): Policy {
+  url = internal_toIriString(url);
```

Comment:
Is this common JS practice (i.e. reassigning an input parameter)? In Java this would be frowned upon (i.e. as a general rule I always use `final` (Java's equivalent of `const`) for all input params).

Reply:
I usually avoid it too, didn't do it here since it's essentially still the same thing and it was on the first line, but I'm definitely up for just banning it altogether (possibly one for https://github.com/inrupt/javascript-style-configs). Fixed: 5ac4efb1971b7eb4bd5142fc2c3e63dddc8ef90a